Automattic/newspack-blocks
# Hide "More" button on private sites

### All Submissions:






### Changes proposed in this Pull Request:



Closes #306.

The "More" button request is returning 403 (Forbidden) when the site is set as `Private` in WP.com settings:

![image](https://user-images.githubusercontent.com/1451471/74753132-f7c5bc00-526f-11ea-82ec-31fed954acaf.png)

Because there's no straightforward way of authenticating the request from the view side of the blog (see https://github.com/Automattic/newspack-blocks/pull/347#issuecomment-585883729), we've decided to hide the button in both the editor and view sides.

### How to test the changes in this Pull Request:

This is a bit complicated because it depends on a WP.com blog property, so I recommend the following:

Editor testing:

1. Create page
2. In the JS console, run `window.wpcomGutenberg.blogPublic = '-1'`
3. Add Newspack Articles block
4. There should not be the `Show "More" button` option available

View testing:

1. Manually set [`$is_blog_private`](https://github.com/Automattic/newspack-blocks/blob/fix/handle-private-blog-load-more/src/blocks/homepage-articles/view.php#L90) to true
2. Publish & go to a page containing the Articles block with the "More" button enabled
3. There should not be the "More" button on the page

### Other information:
